src/BCReader.jl

```diff
@@ -273,9 +273,8 @@ or returns the first Field if interpolation is switched off.
 - Fields.field
 """
 function interpolate_midmonth_to_daily(date, bcf_info::BCFileInfo{FT}) where {FT}
-    if bcf_info.interpolate_daily && bcf_info.segment_length[1] > FT(0)
-        (; segment_length, segment_idx, all_dates, monthly_fields) = bcf_info
-
+    (; segment_length, segment_idx, all_dates, monthly_fields) = bcf_info
+    if bcf_info.interpolate_daily && bcf_info.segment_length[1] > FT(0) && date != all_dates[Int(segment_idx[1])]
```

Comment:
We could extract `interpolate_daily` and use `segment_length` from above, so we don't have to use `bcf_info.<...>`

Reply:
Good point!